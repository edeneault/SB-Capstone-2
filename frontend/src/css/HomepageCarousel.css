@font-face {
  font-family: tesla;
  src: url(./fonts/Suissnord.otf) format("opentype");
}

@font-face {
  font-family: corbel;
  src: url(./fonts/CORBEL.TTF);
}

.carousel {
  height: 25rem;
}

.carousel-inner .carousel-item {
  opacity: 0;
  top: 0;
  left: 0;
  width: 100%;

  display: block;
  position: absolute;
}
.carousel-inner .carousel-item:first-of-type {
  position: relative;
}
.carousel-inner .active {
  opacity: 1;
}
.carousel-caption {
  top: auto;
  bottom: 0;
}

.carousel-text-shadow {
  font-family: corbel;
}

.slide-title {
  font-family: corbel;
  text-shadow: 0px 0px 2px #000000;
}

.img {
}
