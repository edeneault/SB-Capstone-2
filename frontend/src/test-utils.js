import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
