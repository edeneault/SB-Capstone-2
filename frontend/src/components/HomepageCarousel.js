import React from "react";
import { Link } from "react-router-dom";
import { Carousel, Image } from "react-bootstrap";
import "../css/HomepageCarousel.css";

const HomepageCarousel = () => {
  return (
    <Carousel fade className='mt-3 fade-in'>
      <Carousel.Item>
        <Image
          className='img'
          src='https://res.cloudinary.com/edeneault-cloudinary/image/upload/c_fill,h_400,w_2000/v1630186070/eShop/homepage-carousel/homepage-carousel-coffee-2_e6ue2v.jpg'
          alt='First slide'
        />
        <Carousel.Caption>
          <h3 className='text-light fs-md-5 slide-title'>
            Curated Coffee Selection
          </h3>
          <Link
            className='btn btn-secondary rounded slide-btn '
            to='/categories/espresso'
          >
            <span className='slide-in'>SHOP COFFEE</span>
          </Link>
        </Carousel.Caption>
      </Carousel.Item>
      <Carousel.Item>
        <Image
          className='img'
          src='https://res.cloudinary.com/edeneault-cloudinary/image/upload/v1630862474/eShop/homepage-carousel/Carousel-barista_uyypfe.jpg'
          alt='Second slide'
        />

        <Carousel.Caption>
          <h3 className='text-light fs-md-5 slide-title'>
            Barista and Coffee Machines
          </h3>
          <Link
            className='btn btn-secondary rounded slide-btn'
            to='/categories/barista'
          >
            <span className='slide-in'>SHOP MACHINE</span>
          </Link>
        </Carousel.Caption>
      </Carousel.Item>
      <Carousel.Item>
        <Image
          className='img'
          src='https://res.cloudinary.com/edeneault-cloudinary/image/upload/c_fill,h_400,w_2000/v1630862737/eShop/Depositphotos_187848480_sm-2015_ihw3l3.jpg'
          alt='Third slide'
        />

        <Carousel.Caption>
          <h3 className='text-light fs-md-5 slide-title'>
            ACCESSORIES AND MERCHANDISE
          </h3>
          <Link
            className='btn btn-secondary rounded slide-btn'
            to='/categories/accessories'
          >
            <span className='slide-in'>SHOP ACCESSORIES</span>
          </Link>
        </Carousel.Caption>
      </Carousel.Item>
    </Carousel>
  );
};

export default HomepageCarousel;
