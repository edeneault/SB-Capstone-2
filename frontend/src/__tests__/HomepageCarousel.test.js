import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { initialState, render } from "../test-utils";
import HomepageCarousel from "../components/HomepageCarousel.js";

const mockStore = configureMockStore([thunk]);

describe("test", () => {
  test("renders HomepageCarousel without crashing", async () => {
    const store = mockStore(initialState);

    const { getByText } = render(
      <Provider store={store}>
        <HomepageCarousel />
      </Provider>,
    );

    expect(getByText("Curated Coffee Selection")).toBeDefined();
  });

  test("matches snapshot", function () {
    const store = mockStore(initialState);
    const { asFragment } = render(
      <Provider store={store}>
        <HomepageCarousel />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
