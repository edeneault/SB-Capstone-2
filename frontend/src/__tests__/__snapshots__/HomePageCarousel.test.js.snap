// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mt-3 fade-in carousel slide carousel-fade"
  >
    <ol
      class="carousel-indicators"
    >
      <li
        class="active"
      />
      <li />
      <li />
    </ol>
    <div
      class="carousel-inner"
    >
      <div
        class="active carousel-item"
      >
        <img
          alt="First slide"
          class="img"
          src="https://res.cloudinary.com/edeneault-cloudinary/image/upload/c_fill,h_400,w_2000/v1630186070/eShop/homepage-carousel/homepage-carousel-coffee-2_e6ue2v.jpg"
        />
        <div
          class="carousel-caption"
        >
          <h3
            class="text-light fs-md-5 slide-title"
          >
            Curated Coffee Selection
          </h3>
          <a
            class="btn btn-secondary rounded slide-btn "
            href="/categories/espresso"
          >
            <span
              class="slide-in"
            >
              SHOP COFFEE
            </span>
          </a>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <img
          alt="Second slide"
          class="img"
          src="https://res.cloudinary.com/edeneault-cloudinary/image/upload/v1630862474/eShop/homepage-carousel/Carousel-barista_uyypfe.jpg"
        />
        <div
          class="carousel-caption"
        >
          <h3
            class="text-light fs-md-5 slide-title"
          >
            Barista and Coffee Machines
          </h3>
          <a
            class="btn btn-secondary rounded slide-btn"
            href="/categories/barista"
          >
            <span
              class="slide-in"
            >
              SHOP MACHINE
            </span>
          </a>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <img
          alt="Third slide"
          class="img"
          src="https://res.cloudinary.com/edeneault-cloudinary/image/upload/c_fill,h_400,w_2000/v1630862737/eShop/Depositphotos_187848480_sm-2015_ihw3l3.jpg"
        />
        <div
          class="carousel-caption"
        >
          <h3
            class="text-light fs-md-5 slide-title"
          >
            ACCESSORIES AND MERCHANDISE
          </h3>
          <a
            class="btn btn-secondary rounded slide-btn"
            href="/categories/accessories"
          >
            <span
              class="slide-in"
            >
              SHOP ACCESSORIES
            </span>
          </a>
        </div>
      </div>
    </div>
    <a
      class="carousel-control-prev"
      href="#"
      role="button"
    >
      <span
        aria-hidden="true"
        class="carousel-control-prev-icon"
      />
      <span
        class="sr-only"
      >
        Previous
      </span>
    </a>
    <a
      class="carousel-control-next"
      href="#"
      role="button"
    >
      <span
        aria-hidden="true"
        class="carousel-control-next-icon"
      />
      <span
        class="sr-only"
      >
        Next
      </span>
    </a>
  </div>
</DocumentFragment>
`;
